datadog-tracer-flare/src/zip.rs

```diff
@@ -116,10 +116,171 @@ fn zip_files(files: Vec<String>) -> Result<File, FlareError> {
     }
 
     // Finalize the zip
-    zip.finish()
+    let file = zip
+        .finish()
         .map_err(|e| FlareError::ZipError(format!("Failed to finalize zip file: {e}")))?;
 
-    Ok(temp_file)
+    Ok(file)
+}
+
+pub const BOUNDARY: &str = "83CAD6AA-8A24-462C-8B3D-FF9CC683B51B";
+
+/// Generates a multipart form-data payload containing flare information and zip file, including
+/// metadata like source, case ID, hostname, email, UUID and the zip file itself.
+///
+/// # Arguments
+///
+/// * `zip` - The zip file to include
+/// * `language` - Tracer language
+/// * `log_level` - Flare log level
+/// * `case_id` - Agent task case ID
+/// * `hostname` - Agent task hostname
+/// * `user_handle` - Agent task user email
+/// * `uuid` - Agent task UUID
+///
+/// # Returns
+///
+/// * `Ok(Vec<u8>)` - Multipart form-data payload bytes
+/// * `Err(FlareError)` - If zip file read fails
+fn generate_payload(
+    mut zip: File,
+    language: &String,
+    log_level: &String,
+    case_id: &u64,
+    hostname: &String,
+    user_handle: &String,
+    uuid: &String,
+) -> Result<Vec<u8>, FlareError> {
+    let mut payload: Vec<u8> = Vec::new();
+
+    // Create the multipart form data
+    let mut add_part = |name: &str, content: &[u8]| {
```

Comment:
Can't we use an existing library to generate a multipart upload like [hyper-multipart-rfc7578](https://crates.io/crates/hyper-multipart-rfc7578)?

Reply:
It's a pretty simple and straightforward format, so I'd say this is a bit overkill.
This code really is perfectly legible (and multipart uploads neither are a moving target).

Reply:
If this is the extent of the functionality, then that's fine. 

But at a minimum it should be tested.